pkg/network/cluster_config.go

```diff
@@ -39,15 +40,18 @@ func ValidateClusterConfig(clusterConfig configv1.NetworkSpec) error {
 		if err != nil {
 			return errors.Errorf("could not parse spec.clusterNetwork %s", cnet.CIDR)
 		}
-		ones, bits := cidr.Mask.Size()
-		// The comparison is inverted; smaller number is larger block
-		if cnet.HostPrefix < uint32(ones) {
-			return errors.Errorf("hostPrefix %d is larger than its cidr %s",
-				cnet.HostPrefix, cnet.CIDR)
-		}
-		if int(cnet.HostPrefix) > bits-2 {
-			return errors.Errorf("hostPrefix %d is too small, must be a /%d or larger",
-				cnet.HostPrefix, bits-2)
+		// ignore hostPrefix if the plugin does not use it and has it unset
+		if networkTypeRequiresHostPrefix(clusterConfig.NetworkType) || (!networkTypeRequiresHostPrefix(clusterConfig.NetworkType) && cnet.HostPrefix != 0) {
```

Comment:
you don't need the `!networkTypeRequiresHostPrefix()` check since the || condition will get shorted at the first true.

Reply:
oh hey yea you are right, stupid me...